### ISSUE TITLE
When you push pull request to a fork it would be awesome if you could push to master

### ISSUE THREAD

**Issue by skirmess** [peter-evans/create-pull-request#489]
### When you push pull request to a fork it would be awesome if you could push to master 

When you [push pull request branches to a fork](https://github.com/peter-evans/create-pull-request/blob/master/docs/concepts-guidelines.md#push-pull-request-branches-to-a-fork) you cannot push to master (or whatever the branch is you checked out). You get an 
```
 ##[error]Working base branch 'master' was created by this action. Unable to continue.
```
error. But technically this should work because, if you push to a fork, this is the master in a different repo.

The downside with using a machine user is that you have a fork of your repository on github that gets outdated and people might find this instead of the real repository*. If this action could push to master you would at least have an updated master whenever there was something to merge.

### Steps to reproduce

```
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
        with:
          token: ${{ secrets.MACHINE_USER_PAT }}
          push-to-fork: machine-user/repository
          branch: master
```

*) Or do you have an action to keep a fork up to date? :-)

**Comment by peter-evans** [comment]
Hi @skirmess 

You are right, it should be possible to use the default branch in the fork. I will fix the condition to make this possible.
Also, the error message is quite confusing because the original intention was to catch a different problem, which can still occur, but perhaps doesn't warrant an explicit check. I'll update that, too.

**Comment by peter-evans** [comment]
This should now be fixed in `v3.1.1`. Thanks for reporting!
